app/uk/gov/hmrc/advancevaluationrulings/controllers/BaseController.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.advancevaluationrulings.controllers
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+import play.api.libs.json._
+import play.api.mvc.{Request, Result, Results}
+import uk.gov.hmrc.advancevaluationrulings.models.common.Envelope.Envelope
+import uk.gov.hmrc.advancevaluationrulings.models.errors.{ErrorResponse, ValidationError, ValidationErrors}
+import uk.gov.hmrc.http.HeaderCarrier
+
+trait BaseController {
```

Comment:
I wonder if this should be a package object instead since every controller will use this (and would lets us avoid inheritance)

Reply:
I've moved it into the package object. Thanks 👍